conda_smithy/configure_feedstock.py

```diff
@@ -580,7 +583,96 @@ def _load_forge_config(forge_dir, variant_config_files):
+    most_recent_version = available_versions[-1]
+    if installed_version is None:
+        msg = "{} is not installed in root env.".format(name)
+    elif VersionOrder(installed_version) < VersionOrder(most_recent_version):
+        msg = "{} version in root env is out-of-date.".format(name)
+    else:
+        return
+    if error_on_warn:
+        raise RuntimeError("{} Exiting.".format(msg))
+    else:
+        print(msg)
+
+
+def commit_changes(forge_file_directory, commit, cs_ver, cfp_ver):
+    msg = 'Re-rendered with conda-smithy {} and cf-pinning {}'.format(cs_ver, cfp_ver)
+    print(msg)
+
+    is_git_repo = os.path.exists(os.path.join(forge_file_directory, ".git"))
+    if is_git_repo:
+        has_staged_changes = subprocess.call(
+            [
+                "git", "diff", "--cached", "--quiet", "--exit-code"
+            ],
+            cwd=forge_file_directory
+        )
+        if has_staged_changes:
+            if commit:
+                git_args = [
+                    'git',
+                    'commit',
+                    '-m',
+                    'MNT: {}'.format(msg)
+                ]
+                if commit == "edit":
+                    git_args += [
+                        '--edit',
+                        '--status',
+                        '--verbose'
+                    ]
+                subprocess.check_call(
+                    git_args,
+                    cwd=forge_file_directory
+                )
+                print("")
+            else:
+                print(
+                    'You can commit the changes with:\n\n'
+                    '    git commit -m "MNT: {}"\n'.format(msg)
+                )
+            print("These changes need to be pushed to github!\n")
+        else:
+            print("No changes made. This feedstock is up-to-date.\n")
+
+
+def main(forge_file_directory, variant_config_files, no_check_uptodate, commit):
+    error_on_warn = False if no_check_uptodate else True
+    index = conda_build.conda_interface.get_index(channel_urls=['conda-forge']) 
+    r = conda_build.conda_interface.Resolve(index)
+
+    # Check that conda-smithy is up-to-date
+    check_version_uptodate(r, "conda-smithy", __version__, error_on_warn)
+
+    # Don't check for conda-forge-pinning if variant_config_files is given
+    if variant_config_files is None:
+        cf_pinning_ver = get_installed_version("conda-forge-pinning")
+        if cf_pinning_ver:
+            check_version_uptodate(r, "conda-forge-pinning", cf_pinning_ver, error_on_warn)
+        else:
+            raise RuntimeError("Install conda-forge-pinning or give a config file using -m")
+        config_file = os.path.join(conda_build.conda_interface.root_dir, "conda_build_config.yaml")
+        if not os.path.exists(config_file):
+            raise RuntimeError("conda-build_config.yaml from conda-forge-pinning is missing")
+        variant_config_files = [config_file]
```

Comment:
So I think this should be passed as `exclusive_config_file`. PR ( https://github.com/conda/conda-build/pull/2661 ) might help make sense of this a bit. Basically we are trying to make sure that user's local settings don't clobber the re-rendering. Setting `exclusive_config_file` as `conda_build_config.yaml` should do that. In addition, we can add other variant files.

Though am less clear if `conda-build` needs us to provide those variant files if they are already in the recipe directory or if it will detect them automatically. Maybe @msarahan can comment on this.